### PR TITLE
Add semantic-release

### PR DESCRIPTION
**Pull request by n-rodriguez**
Before moving to a new implementation of this formula : https://github.com/saltstack-formulas/influxdb-formula/issues/23#issuecomment-524331438

### CODE REVIEW

**Comment by myii** [comment]
@n-rodriguez Happy to review when ready.  Please ping me again when it's ready to go.

**Comment by myii** [comment]
@n-rodriguez Also, would you mind using the latest platform matrix from the `template-formula`?

**Comment by aboe76** [comment]
@n-rodriguez nice work, looking good, the more formulas get semantic release the better.

**Comment by n-rodriguez** [comment]
@myii @aboe76 I think we're ready to merge :) https://travis-ci.com/saltstack-formulas/influxdb-formula

**Comment by myii** [comment]
@n-rodriguez Thanks, I'll have a look in the next few hours.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#       testing of the formula's behaviour. If possible, try to refrain from
+#       the classical "chosing all the instances because I want to test on
+#       another/all distro/s" trap: it will just add time to the testing (see
+#       the discussion on #121).  As an example, the set chosen below covers
+#       the most used distros families, systemd and non-systemd and the latest
+#       three supported Saltstack versions with python2 and 3.
+#       As for `kitchen.yml`, that should still contain all of the platforms,
+#       to allow for comprehensive local testing
+#       Ref: https://github.com/saltstack-formulas/template-formula/issues/118
+#       Ref: https://github.com/saltstack-formulas/template-formula/issues/121
```
Comment (myii) [inline .travis.yml]: Please remove this 11-line `NOTE`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+env:
+  matrix:
+    # - INSTANCE: default-debian-9-develop-py3
```
Comment (myii) [inline .travis.yml]:
```suggestion
    - INSTANCE: default-debian-9-develop-py3
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    # - INSTANCE: default-ubuntu-1804-develop-py3
+    # - INSTANCE: default-centos-7-develop-py3
+    # - INSTANCE: default-fedora-29-develop-py3
+    # - INSTANCE: default-opensuse-leap-15-develop-py3
+    - INSTANCE: default-debian-9-2019-2-py3
```
Comment (myii) [inline .travis.yml]:
```suggestion
    # - INSTANCE: default-debian-9-2019-2-py3
```

File: docs/TOFS_pattern.rst [hunk 4]
```diff
@@ -0,0 +1,446 @@
+.. _tofs_pattern:
```
Comment (myii) [inline docs/TOFS_pattern.rst]: This file isn't needed for now.

File: test/integration/default/inspec.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (myii) [inline test/integration/default/inspec.yml]:
```suggestion
# -*- coding: utf-8 -*-
# vim: ft=yaml
---
```

File: docs/README.rst [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+``influxdb``
+^^^^^^^^^^^^
+Installs InfluxDB from [provided packages](http://influxdb.com/download/)
+
+At the moment you **have** to specify the InfluxDB version in the
+``influxdb:version`` pillar.
+
+``influxdb.cli``
+^^^^^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^
```

File: docs/README.rst [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+.. code-block:: bash
+   $ gem install bundler
+   $ bundle install
+   $ bin/kitchen test [platform]
+Where ``[platform]`` is the platform name defined in ``kitchen.yml``,
+e.g. ``debian-9-2019-2-py3``.
+
+
+``bin/kitchen converge``
+^^^^^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/README.rst [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+Creates the docker instance and runs the ``influxdb`` main state, ready for testing.
+
+``bin/kitchen verify``
+^^^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/README.rst [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+Runs the ``inspec`` tests on the actual instance.
+
+``bin/kitchen destroy``
+^^^^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/README.rst [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+Removes the docker instance.
+
+``bin/kitchen test``
+^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^
```

File: docs/README.rst [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+Runs all of the stages above in one go: i.e. ``destroy`` + ``converge`` + ``verify`` + ``destroy``.
+
+``bin/kitchen login``
+^^^^^^^^^^^^^^^^^
```
Comment (myii) [inline docs/README.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^
```

**Comment by myii** [comment]
@n-rodriguez I left out `yamllint` changes that are now being used across our `semantic-release` formulas (see https://github.com/saltstack-formulas/template-formula/pull/159).  Shall we leave them for a subsequent PR or are you OK with me adding a commit to this PR?

In terms of `test/salt/default/pillar/influxdb.sls`, that sort of path is used in a few of our formulas but the majority of our formulas would use `test/salt/pillar/influxdb.sls`.  How do you feel about changing that?

**Comment by n-rodriguez** [comment]
>  How do you feel about changing that?

I'm ok with that.

**Comment by n-rodriguez** [comment]
> Shall we leave them for a subsequent PR or are you OK with me adding a commit to this PR?

Feel free to add a commit

**Comment by myii** [comment]
@n-rodriguez It's all merged, thanks for another important PR.  I added another commit for `yamllint` and I also made tiny modifications to the `README`, so that it matches the `template-formula` version -- there are some changes that need to be propagated soon, so it makes it easier if everything lines up.

**Pull request merged**